bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/editor/targetComponent/details/SiderealNameEditor.scala

```diff
@@ -0,0 +1,143 @@
+package jsky.app.ot.gemini.editor.targetComponent.details
+
+import java.awt.{Insets, GridBagConstraints, GridBagLayout}
+import javax.swing.JPanel
+
+import edu.gemini.pot.sp.ISPNode
+import edu.gemini.shared.util.immutable.{ Option => GOption }
+import edu.gemini.catalog.skycat.CatalogException
+import edu.gemini.spModel.obs.context.ObsContext
+import edu.gemini.spModel.target.SPTarget
+import edu.gemini.spModel.target.system.CoordinateParam.Units
+import edu.gemini.spModel.target.system.HmsDegTarget
+import jsky.app.ot.gemini.editor.targetComponent.TelescopePosEditor
+import jsky.catalog.skycat.{SkycatConfigFile, FullMimeSimbadCatalogFilter, SkycatCatalog}
+import jsky.catalog.{BasicQueryArgs, TableQueryResult, Catalog}
+import jsky.coords.WorldCoords
+import jsky.util.gui.{DialogUtil, DropDownListBoxWidget, TextBoxWidgetWatcher, TextBoxWidget}
+
+import scalaz.concurrent.Task
+import scalaz.syntax.id._
+import scalaz.{ \/-, -\/ }
+
+// Name editor, with catalog lookup for sidereal targets
+final class SiderealNameEditor extends JPanel with TelescopePosEditor with ReentrancyHack {
+
+  private[this] var spt = new SPTarget // never null
+
+  val name = new TextBoxWidget <| { w =>
+    w.setColumns(20)
+    w.setMinimumSize(w.getPreferredSize)
+    w.addWatcher(new TextBoxWidgetWatcher {
+
+      override def textBoxKeyPress(tbwe: TextBoxWidget): Unit =
+        nonreentrant {
+          spt.getTarget.setName(tbwe.getValue)
+          spt.notifyOfGenericUpdate()
+        }
+
+      override def textBoxAction(tbwe: TextBoxWidget): Unit =
+        lookupTarget(cats.getSelectedItem.asInstanceOf[Catalog], tbwe.getValue).runAsync {
+          case \/-(r) => processResult(r)
+          case -\/(e) => DialogUtil.error(e.getMessage)
+        }
+
+    })
+  }
+
+  val cats = new DropDownListBoxWidget <| { m =>
+    m.setChoices(SkycatConfigFile.getConfigFile.getNameServers)
+  }
+
+  setLayout(new GridBagLayout)
+
+  add(name, new GridBagConstraints <| { c =>
+    c.gridx = 0
+    c.gridy = 0
+    c.fill = GridBagConstraints.HORIZONTAL
+    c.weightx = 2
+  })
+
+  add(cats, new GridBagConstraints <| { c =>
+    c.gridx = 1
+    c.gridy = 0
+    c.insets = new Insets(0, 2, 0, 0)
+  })
+
+  def edit(ctx: GOption[ObsContext], target: SPTarget, node: ISPNode): Unit = {
+    this.spt = target
+    nonreentrant {
+      name.setText(spt.getTarget.getName)
+    }
+  }
+
+  def lookupTarget(catalog: Catalog, name: String): Task[TableQueryResult] =
```

Comment:
It's a pity that we are still doing code against the old SkyCatalog code, the plan is to remove it, but name servers are still on the todo list